src/urls.rs

```diff
@@ -69,3 +69,61 @@ pub fn is_url(input: &str) -> bool {
                                   (input.contains('.') || input.contains(':'))) ||
         input == "localhost"
 }
+
+/// Take url and increment the first number with offset    
+pub fn offset(url: &str, inc_offset: i32) -> Option<String> {
+    if let Ok(url) = Url::parse(url) {
+        let mut updated = false;
+
+        if let Some(query) = url.query(){
+            let pairs = url.query_pairs().into_owned();
+
+            // ParseIntoOwned lacks DoubleEndedIterator for rev(), must be parsed left to right
+            let next = pairs.map(|(lhs, rhs)| {
+                if updated {
+                    lhs + "=" + &rhs
+                } else if let Ok(number) = rhs.parse::<i32>() {
+                    updated = true;
+                    lhs + "=" + (number + inc_offset).to_string().as_str()
```

Comment:
This is going to increment every number in the URL query string, right?
That might not be what we want.